### PR TITLE
Add documentation on feature branches on Concourse per skyscrapers/engineering#292

### PR DESCRIPTION
**Pull request by simonrondelez**

### CODE REVIEW

File: Concourse/feature_environments.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
+# Feature environments
+
+At this moment Its not yet possible to handle feature environments or as Concourse calls it: `Spaces` per default in Concourse. Concourse is working hard towards having Spaces out of the box in Concourse but that will require some time.
```
Comment (duboisph) [inline Concourse/feature_environments.md]: Its -> it's

having Spaces out of the box in Concourse --> having Spaces available out of the box, but that ...

**Comment by simonrondelez** [comment]
for the documentation on this i sticked to the basic concept of having feature pipelines as the specific implementation of it will be different per use case

File: Concourse/feature_environments.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+With this documentation we give an example on a way that you can have a similar experience with Concourse right now.
```
Comment (duboisph) [inline Concourse/feature_environments.md]: on a way that -> on how you can have ...

File: Concourse/feature_environments.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+export OLD_VERSIONS=$(cat puppet/removed)
+
+fly login -t skyscrapers --team-name skyscrapers -c https://ci.example.com -u -p $CONCOURSE_PASSWORD
+fly -t skyscrapers sync
+fly login -t skyscrapers --team-name skyscrapers -c https://ci.example.com -u -p $CONCOURSE_PASSWORD
+
+for version in $NEW_VERSIONS; do
+  sed "s/___BRANCH___/$version/g" ci-git/puppet_multiBranch/pipeline-puppet.tmpl > ./pipeline-puppet.result
+  echo "Create pipeline branch $version"
+  fly -t skyscrapers sp -n -p puppet-$version -c ./pipeline-puppet.result
```
Comment (duboisph) [inline Concourse/feature_environments.md]: General remark: it usually helps to use the long command versions (like `set-pipelines`) in the docs for readers to better understand what's going on.

**Pull request merged**